guardian/actions-riff-raff
# feat!: Replace `sources` with `contentDirectories`

Fixes https://github.com/guardian/actions-riff-raff/issues/13.

> **Note**
> This will need a new major version.
> Updates to the README can be more easily viewed [here](https://github.com/guardian/actions-riff-raff/blob/aa-sources/README.md).

## What does this change?



In this change we remove support for `sources` in the body of the `riff-raff.yaml` config, and replace it with a `contentDirectories` input to the Action.

This decoupling keeps the `riff-raff.yaml` config "valid". That is, we're able to copy/paste it to [Riff-Raff's validation tool](https://riffraff.gutools.co.uk/configuration/validation) w/out first needing to remove `sources`.

It also means this action becomes compatible with auto-generated `riff-raff.yaml` files.

⚠️  BREAKING CHANGE: `sources` within `riff-raff.yaml` (either via the `config` or `configPath` input) is no longer supported

Configure the files to upload via the `contentDirectories` input, where `contentDirectories` is a mapping of package name -> artifact files.

From:

```yaml
- uses: guardian/actions-riff-raff@v1
  with:
    app: my-application
    config: |
      stacks:
        - deploy
      regions:
        - eu-west-1
      allowedStages:
        - CODE
        - PROD
      deployments:
        static-site-assets:
          type: aws-s3
          sources:
            - static-site/dist # upload the `dist` directory
          parameters:
            bucket: aws-some-bucket
            cacheControl: private
            publicReadAcl: false
        api:
          type: autoscaling
          sources:
            - target/application.jar
          parameters:
            bucketSsmLookup: true
```

To:

```yaml
- uses: guardian/actions-riff-raff@v2 # Note the version change
  with:
    app: my-application
    config: |
      stacks:
        - deploy
      regions:
        - eu-west-1
      allowedStages:
        - CODE
        - PROD
      deployments:
        static-site-assets: # <-- this is a package name
          type: aws-s3
          parameters:
            bucket: aws-some-bucket
            cacheControl: private
            publicReadAcl: false
        api: # <-- this is another package name
          type: autoscaling
          parameters:
            bucketSsmLookup: true
    contentDirectories: |
      static-site-assets:
        - static-site/dist
      api:
        - target/application.jar
```

## How to test



See:
  - https://github.com/guardian/cdk-playground/pull/284
  - https://github.com/guardian/tracker/pull/551
